io/genbank/genbank.go

```diff
@@ -146,222 +144,551 @@ func getFeatureSequence(feature Feature, location Location) (string, error) {
+				}
+
+				// add our features to the genbank
+				for _, feature := range parameters.features {
+					err := parameters.genbank.AddFeature(&feature)
+					if err != nil {
+						return []Genbank{}, err
+					}
+				}
+				continue
+			} // end sequence parsing flag logic
+
+			// If there is no active quote and the line does not have a / and newLocation == false, we know this is a top level feature.
+			if !parameters.quoteActive && !strings.Contains(line, "/") && !parameters.newLocation {
+				// Check for empty types
+				if parameters.feature.Type != "" {
+					var err error
+					parameters.feature.Location, err = parseLocation(parameters.feature.Location.GbkLocationString)
+					if err != nil {
+						return []Genbank{}, err
+					}
+					parameters.features = append(parameters.features, parameters.feature)
+				}
 
-	// start writing sequence section.
-	gbkString.WriteString("ORIGIN\n")
+				parameters.feature = Feature{}
+				parameters.feature.Attributes = make(map[string]string)
 
-	// iterate over every character in sequence range.
-	for index, base := range sequence.Sequence {
-		// if 60th character add newline then whitespace and index number and space before adding next base.
-		if index%60 == 0 {
-			if index != 0 {
-				gbkString.WriteString("\n")
+				// An initial feature line looks like this: `source          1..2686` with a type separated by its location
+				if len(splitLine) < 2 {
+					return genbanks, fmt.Errorf("Feature line malformed on line %d. Got line: %s", lineNum, line)
+				}
+				parameters.feature.Type = strings.TrimSpace(splitLine[0])
+				parameters.feature.Location.GbkLocationString = strings.TrimSpace(splitLine[len(splitLine)-1])
+
+				// Set parameters.newLocation = true to so that we can check the next line for a multi-line location string
+				parameters.newLocation = true
+				continue
+			} // end top level feature logic
+
+			// If not a newFeature, check if the next line does not contain "/". If it does not, then it is a multi-line location string.
+			if !strings.Contains(line, "/") && parameters.newLocation {
+				parameters.feature.Location.GbkLocationString = parameters.feature.Location.GbkLocationString + strings.TrimSpace(line)
+				continue
 			}
-			lineNumberString := strconv.Itoa(index + 1)          // genbank indexes at 1 for some reason
-			leadingWhiteSpaceLength := 9 - len(lineNumberString) // <- I wish I was kidding
-			for i := 0; i < leadingWhiteSpaceLength; i++ {
-				gbkString.WriteString(" ")
+
+			// this smells bad
+			parameters.newLocation = false
+
+			// First, let's check if we have a quote active (basically, if a attribute is using multiple lines)
+			if parameters.quoteActive {
+				trimmedLine := strings.TrimSpace(line)
+				if len(trimmedLine) < 1 {
+					continue
+				}
+				// If the trimmed line ends, append to the attribute and set parameters.quoteActive set to false
+				if trimmedLine[len(trimmedLine)-1] == '"' {
```

Comment:
this won't work with all labels. Labels in t4_intron.gb don't have quotes surrounding label string.

Reply:
attention @Koeng101 can you work on this bug? Essentially you can't rely on quotations to show you the end of an attribute because some one line attributes don't use any quotations like this:

```
 gene            1..243
                 /gene="frd.1"
                 /locus_tag="T4p240"
                 /label=frd.1  <--- offending line
                 /db_xref="GeneID:1258741"

```